### PR TITLE
Fix sampler nan check when calling top_k_top_p_sampling_from_probs

### PR DESCRIPTION
**Pull request by yubofredwang**
## Motivation

~~Remove success tensor that is not returned from top_k_top_p_sampling_from_probs of the latest flash infer api anymore.~~

The current code on main branch will throw variable referenced before assigned error if `self.use_nan_detection` evaluate to true. This PR fixes it.

We don't need to check for success anymore as the PR in flash infer has improved the robustness of the sampling. Now tokens are guaranteed to be generated.
Details in: https://github.com/flashinfer-ai/flashinfer/pull/912

Please refer to flash infer for reference:
https://github.com/flashinfer-ai/flashinfer/blob/main/flashinfer/sampling.py

## Modifications

Remove successor, pass in check for nan flag, and fix the function signature.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

## Test Result
No latency regression after the change.

```python3 -m unittest test_bench_one_batch.TestBenchOneBatch.test_bs1```

Before change:
```
Warmup ...
Prefill. latency: 0.33793 s, throughput:    378.77 token/s
Decode.  median latency: 0.00587 s, median throughput:    170.22 token/s
Total. latency:  0.764 s, throughput:    178.08 token/s
Benchmark ...
Prefill. latency: 0.01321 s, throughput:   9686.79 token/s
Decode.  median latency: 0.00580 s, median throughput:    172.50 token/s
Total. latency:  0.054 s, throughput:   2526.44 token/s
```
After change:
```
Warmup ...
Prefill. latency: 0.50365 s, throughput:    254.14 token/s
Decode.  median latency: 0.00589 s, median throughput:    169.79 token/s
Total. latency:  1.031 s, throughput:    131.89 token/s
Benchmark ...
Prefill. latency: 0.01323 s, throughput:   9676.49 token/s
Decode.  median latency: 0.00579 s, median throughput:    172.60 token/s
Total. latency:  0.054 s, throughput:   2527.25 token/s
```

### CODE REVIEW

**Comment by FlamingoPg** [comment]
U can fix sgl-kernel/python/sgl_kernel/sampling.py firstly, for python/sglang/srt/layers/sampler.py we need bump sgl-kernel to new version.

**Comment by yubofredwang** [comment]
> U can fix sgl-kernel/python/sgl_kernel/sampling.py firstly, for python/sglang/srt/layers/sampler.py we need bump sgl-kernel to new version.

Yes, but the change in sgl_kernel is not required for the code to work. python does not do type checking. so returning `torch.Tensor` or a Tuple won't affect code execution.

This change is just to keep things consistent with flash infer

File: python/sglang/srt/layers/sampler.py [hunk 1]
```diff
@@ -100,17 +100,16 @@ def forward(
...
+                    check_nan = self.use_nan_detection and crash_on_warnings()
                     batch_next_token_ids = top_k_top_p_sampling_from_probs(
                         probs,
                         sampling_info.top_ks,
                         sampling_info.top_ps,
                         filter_apply_order="joint",
+                        check_nan=check_nan,
                     )
 
-                    if self.use_nan_detection:
```
Comment (yubofredwang) [inline python/sglang/srt/layers/sampler.py:110]: Adding a comment here as well: we can remove this check as the new flash infer sampling implementation guarantees  the generation of enough tokens. See PR: https://github.com/flashinfer-ai/flashinfer/pull/912

File: python/sglang/srt/layers/sampler.py [hunk 2]
```diff
@@ -100,17 +100,16 @@ def forward(
                         probs, sampling_info.min_ps
                     )
                 else:
+                    # Check Nan will throw exception, only check when crash_on_warnings is True
+                    check_nan = self.use_nan_detection and crash_on_warnings()
```
Comment (yubofredwang) [inline python/sglang/srt/layers/sampler.py:104]: Adding another comment for future reference: because we already check for Nan for logits before they get converted into probs. The chance of probs being Nan is quite low. Even when probs contain NaN, sampling was still able to generate tokens

**Review by zhyncs — APPROVED** [review]
